R/call_slack_internals.R

```diff
@@ -0,0 +1,150 @@
+  with_pagination(
+    function() {
+      call_slack_api(
+        "/api/conversations.list",
+        .method = GET,
+        types = types,
+        ...
+      )
+    },
+    extract = "channels"
+  )
+}
+
+
+#' Lists all users in a Slack team.
+#'
+#' @inheritParams auth_test
+#' @keywords internal
+#' @noRd
+#' @references https://api.slack.com/methods/users.list
+list_users <- function(bot_user_oauth_token = Sys.getenv("SLACK_BOT_USER_OAUTH_TOKEN"), ...) {
+  with_pagination(
+    function() {
+      call_slack_api(
+        "/api/users.list",
+        .method = GET,
+        ...
+      )
+    },
+    extract = "members"
+  )
+}
+
+#' Sends a message to a channel.
+#'
+#' @inheritParams auth_test
+#' @keywords internal
+#' @noRd
+#'
+#' @param txt Passed to `text` parameter of `chat.postMessage` API
+#' @param channel Passed to `channel` parameter of `chat.postMessage` API
+#' @param username Passed to `username` parameter of `chat.postMessage` API
+#' @param as_user Passed to `as_user` parameter of `chat.postMessage` API
+#' @param link_names Passed to `link_names` parameter of `chat.postMessage` API
+#'
+#' @references https://api.slack.com/methods/chat.postMessage
+post_message <- function(
+  txt,
+  channel,
+  username = Sys.getenv("SLACK_USERNAME"),
+  bot_user_oauth_token = Sys.getenv("SLACK_BOT_USER_OAUTH_TOKEN"),
+  ...)
+{
+  z <-
+    call_slack_api(
+    "/api/chat.postMessage",
+    .method = POST,
+    body = list(
+      text = txt,
+      channel = slackr_chtrans(channel),
+      username   = username,
+      as_user = TRUE,
+      link_names = 1,
+      ...
+    )
+  )
+  invisible(content(z))
+}
+
+
+#' Sends a message to a channel.
+#'
+#' @inheritParams auth_test
+#' @keywords internal
+#' @noRd
+#'
+#' @param txt Passed to `text` parameter of `chat.postMessage` API
+#' @param channel Passed to `channel` parameter of `chat.postMessage` API
+#' @param username Passed to `username` parameter of `chat.postMessage` API
+#' @param as_user Passed to `as_user` parameter of `chat.postMessage` API
+#' @param link_names Passed to `link_names` parameter of `chat.postMessage` API
+#'
+#' @references https://api.slack.com/methods/chat.postMessage
+post_message <- function(
+  txt,
+  channel,
+  username = Sys.getenv("SLACK_USERNAME"),
+  bot_user_oauth_token = Sys.getenv("SLACK_BOT_USER_OAUTH_TOKEN"),
+  ...)
+{
+  z <- call_slack_api(
+    "/api/chat.postMessage",
+    .method = POST,
+    body = list(
+      text = txt,
+      channel = slackr_chtrans(channel),
+      username   = username,
+      as_user = TRUE,
+      link_names = 1,
+      ...
+    )
+  )
+  invisible(content(z))
+}
+
+
+#' Uploads or creates a file.
+#'
+#' This needs the scope `files:write:user`
+#'
+#' @inheritParams auth_test
+#' @inheritParams post_message
+#'
+#' @param file Name of file to upload
+#'
+#' @keywords internal
+#' @noRd
+#'
+#'
+#' @references https://api.slack.com/methods/files.upload
+files_upload <- function(
```

Comment:
Will this one supercede `slackr_upload()` and `save_slackr()`?

Reply:
No. This is an internal function that will be used by many other functions. At the moment I've only tested this with `ggslackr` but it should also work with `dev_slackr()` and eventually these functions you mention.

At the moment I'm still trying to not change the exported API.